### ISSUE TITLE
Request: add scale=<float> in local style.conf with a shared style.conf in Notebook folder

### ISSUE THREAD

**Issue by TammyJess** [zim-desktop-wiki/zim-desktop-wiki#1204]
**Background:** I share a single notebook hiearchy from a nfs share to multiple computers.  It's my memory/external brain ... so I don't have to remember everything.  Zim is a godsend!  Yes, the notebook gets archived, I don't want a hardware-based lobotomy.

**Issue:** different computers have different dpi monitors attached, e.g hiDPI Laptop/Den/Workstation/lowDPI Laptop/etc. Currently the styles are adjusted in the ~/.config/zim/style.conf, so *everything* must be re-adjusted to suit the local conditions.

**Request:** I would like suggest simplifying the adjustment of the visual size between each local computer, regardless whether viewing on a HiDPI monitor or a lowDPI one. 11pt is too small on one, perfect on another.

**Proposed solution:**  It might be easier if each Notebook instance used the hidden .zim folder to store a style.conf with the 'global' styles, and the local style.conf could then override the established (if any) globals, e.g. a local style.conf could contain just `scale=1.5`. This could also be used to adjust colours if one computer uses a dark theme, another uses a light theme.  

**Comment:** I'm happy to keep on as I am as it isn't too arduous, but just occasionally* I think "there has to be a better way".  

*Especially when changing the zoom, and finding that 'View'>'Normal Size' resets the default size to 9pt.

**Comment by jaap-karssenberg** [comment]
Maybe I don't fully understand the issue. If you set the font in the
preferences dialog once for each system, doesn't it work fine?

Or are you looking for a way to share a custom style?

Op zo 26 jul. 2020 07:06 schreef TammyJess <notifications@github.com>:

**Comment by TammyJess** [comment]
Yes, I am trying to share a custom style for this notebook (current 'flavour' is Green  :) , but also want to adjust the overall rendering to correct for a one system with a 96dpi monitor, and a 134dpi monitor on a laptop.

I've noticed in my style.conf that I'm assigning absolute point-sizes and resetting the already present 'scale' to 1.0, so perhaps the new attribute would be better named ` 'dpi-factor': Float(None)`    
Obviously I didn't like the size^4, size^3, size^2, size scheme, and hard coded the sizes in my conf file.  (Sorry!)

For example, in Notebook/.zim/style.conf
```
[Tag h2]
scale=1.0
font=Akzidenz-Grotesk Pro Bold Ext 18
foreground=#1BB04C
linespacing=4

[TextView]
linespacing=2
font=Gotham Rounded Light 12.5
bullet_icon_size=GTK_ICON_SIZE_LARGE_TOOLBAR
```
and in ~/.config/zim/style.conf  (on HiDPI monitor) all I would need to adjust is:
```
dpi-factor=1.3

[Tag h2]
foreground=#2738C1
```
Um, that raises an interesting thought... maybe the 'Preferences'>'Use a Custom Font' should dictate the dpi-factor, and I should use `scale=Float` in the conf file to adjust the relative sizes to my pleasure.  This, however, wouldn't remove the need to copy the custom style onto each machine though.
It could also be why 'View'>'Normal Size' seems to drag things back to the 9pt size.  9pt is the value set in XFCE4's 'Appearance' >'Fonts' pane, with a different hard size (12.5pt) set in the zim style.conf file. Maybe things are getting confused?

**Comment by TammyJess** [comment]
Confirmed.
'View'>'Normal Size' picks up the point-size as set in XFCE4's 'Appearance' >'Fonts' pane, not the size set out in [TextView]. 
As far as I'm concerned this is not an expected behaviour.  Tentatively, it's a bug. Worse, it stomps on the point-size set in the style.conf file.
Unchecking 'Preferences'>'Use a Custom Font' also wipes all the font information set in style.conf. Also unexpected.
My guess is the code is grabbing the Gtk default font details and using one or both the attributes, (name)(size), rather than flagging to use (or not) the style.conf values. 
Surely the behaviour of 'View'>'Normal Size' should just be to just remove any 'Zoom In'/'Zoom Out' overrides, by reloading/reapplying the style.conf file?
Perhaps some of confusion can be removed by differentiating them both, calling what I'm trying to achieve a 'theme' file?
It could still facilitate the style.conf overriding parts of the theme-file, with the use of dpi-factor.. and if a light and dark theme are both created in the Notebook folder (or even a set of themes à là geany, xed, etc), the style.conf could reference these per notebook or system as well?

**Comment by introt** [comment]
Slightly related: Allow float in font size #1018

**Comment by introt** [comment]
> Issue: different computers have different dpi monitors attached, e.g hiDPI Laptop/Den/Workstation/lowDPI Laptop/etc. Currently the styles are adjusted in the ~/.config/zim/style.conf, so everything must be re-adjusted to suit the local conditions.
>
> Request: I would like suggest simplifying the adjustment of the visual size between each local computer, regardless whether viewing on a HiDPI monitor or a lowDPI one. 11pt is too small on one, perfect on another.

Perhaps I'm misunderstanding something, but shouldn't Zim's font size be the same amount of points on every system, while the OS does its magic to calculate the appropriate rendering sizes per screen? In other words, this sounds like something that should be solved by the desktop environment using per display fractional scaling settings, not implemented by individual apps.
